cmd/system-probe/config/config_test.go

```diff
@@ -95,3 +96,44 @@ func TestEventStreamEnabledForSupportedKernelsWindowsUnsupported(t *testing.T) {
 		require.False(t, cfg.GetBool("event_monitoring_config.network_process.enabled"))
 	})
 }
+
+func TestEnableDiscovery(t *testing.T) {
+	t.Run("via YAML", func(t *testing.T) {
+		config.ResetSystemProbeConfig(t)
+		cfg := configurationFromYAML(t, `
+discovery:
+  enabled: true
+`)
+		assert.True(t, cfg.GetBool(discoveryNS("enabled")))
+	})
+
+	t.Run("via ENV variable", func(t *testing.T) {
+		config.ResetSystemProbeConfig(t)
+		t.Setenv("DD_DISCOVERY_ENABLED", "true")
+		cfg := config.SystemProbe
+
+		assert.True(t, cfg.GetBool(discoveryNS("enabled")))
+	})
+
+	t.Run("default", func(t *testing.T) {
+		config.ResetSystemProbeConfig(t)
+		cfg := config.SystemProbe
+
+		assert.False(t, cfg.GetBool(discoveryNS("enabled")))
+	})
+}
+
+func configurationFromYAML(t *testing.T, yaml string) config.Config {
+	f, err := os.CreateTemp(t.TempDir(), "system-probe.*.yaml")
+	require.NoError(t, err)
+	defer os.Remove(f.Name())
+
+	b := []byte(yaml)
+	n, err := f.Write(b)
+	require.NoError(t, err)
+	require.Equal(t, len(b), n)
+	f.Sync()
```

Comment:
missing `f.Close()`

Reply:
I think you need the `Close` call right after the `Sync` call, otherwise you are going to try to remove the file while there is still an open FD.

Reply:
That or just remove the `os.Remove` call altogether, because using `t.TempDir()` means it will be removed.